htdocs/core/actions_massactions.inc.php

```diff
@@ -1328,15 +1328,18 @@
 		}
 	}
 
-	if (!$error) {
-		if ($nbok > 1) {
-			setEventMessages($langs->trans("RecordsDeleted", $nbok), null, 'mesgs');
-		} else {
-			setEventMessages($langs->trans("RecordDeleted"), null, 'mesgs');
-		}
-		$db->commit();
-	} else {
-		$db->rollback();
+	// Message for elements well deleted
+	if ($nbok > 1) {
+		setEventMessages($langs->trans("RecordsDeleted", $nbok), null, 'mesgs');
+	} elseif (!empty($nbok)) {
+		setEventMessages($langs->trans("RecordsDeleted", '1'), null, 'mesgs');
+	}
+	$db->commit();
```

Comment:
If an error occured during a massaction, we must not make a commit. For example a trigger or an external module may have failed in doing a complementary action like deleting into a synchronized system, in such case, we must not validate the deletion into dolibarr.